### PR TITLE
fix: call addSharedMetadataHeaders even when token has not expired

### LINKED ISSUE #1114: x-goog-user-project header has gone in second request
#### Environment details

  - OS: macOS BigSur 11.1
  - Node.js version: v14.15.3
  - npm version: 6.14.10
  - `google-auth-library` version: 6.1.4

#### Steps to reproduce

1. `gcloud auth application-default set-quota-project MY_PROJECT_ID`
2. 
```
export const authTest = async () => {
  const auth = new GoogleAuth();
  const client = await auth.getClient();
  const headersFirst = await client.getRequestHeaders();
  console.log(headersFirst);
  const headersSecond = await client.getRequestHeaders();
  console.log(headersSecond);
}
authTest();
```
3.
```
{
  Authorization: 'Bearer xxxx',
  'x-goog-user-project': 'MY_PROJECT_ID'
}
{
  Authorization: 'Bearer xxxx'
}
```

As a result, I got the following error in **second** request.
```
    7 PERMISSION_DENIED: Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the datacatalog.googleapis.com. We recommend configuring the billing/quota_project setting in gcloud or using a service account through the auth/impersonate_service_account setting. For more information about service accounts and how to use them in your application, see https://cloud.google.com/docs/authentication/.
```

Any help would be appreciated.

### PR DESCRIPTION
**Pull request by hideto0710**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-auth-library-nodejs/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1114  🦕

---

```
npm test -- -g "getRequestHeaders should populate x-goog-user-project"
```
before
```
  refresh
    ✓ getRequestHeaders should populate x-goog-user-project header if quota_project_id present
    1) getRequestHeaders should populate x-goog-user-project header if quota_project_id present and token has not expired


  1 passing (43ms)
  1 failing

  1) refresh
       getRequestHeaders should populate x-goog-user-project header if quota_project_id present and token has not expired:
     AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ undefined
- 'my-quota-project'

```
after
```
  refresh
    ✓ getRequestHeaders should populate x-goog-user-project header if quota_project_id present
    ✓ getRequestHeaders should populate x-goog-user-project header if quota_project_id present and token has not expired
```

### CODE REVIEW

**Comment by hideto0710** [comment]
@googlebot I signed it!

**Comment by bcoe** [comment]
@hideto0710 good catch 👍 (CC: @sofisl). If I'm following correctly, the quota project was properly populated up until the token refreshed?

File: test/test.refresh.ts [hunk 1]
```diff
@@ -152,4 +152,22 @@ describe('refresh', () => {
     assert.strictEqual(headers['x-goog-user-project'], 'my-quota-project');
     req.done();
   });
+
+  it('getRequestHeaders should populate x-goog-user-project header if quota_project_id present and token has not expired', async () => {
```
Comment (bcoe) [inline test/test.refresh.ts:156]: I think we should also add a test that makes sure you get the appropriate header populated after the expiry?

(_sorry for the slow review, we'll work on getting this out the door quickly._).

**Review by bcoe — CHANGES_REQUESTED** [review]
mind adding the additional test? thank you very much for the contribution 🎉

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@hideto0710 thank you very much for the contribution 👍

**Pull request merged**